packages/next/src/utils.ts

```diff
@@ -2255,15 +2254,20 @@ export const onPrerenderRoute =
       const lambdaId = pageLambdaMap[outputSrcPathPage];
       lambda = lambdas[lambdaId];
     } else {
-      const outputSrcPathPage = normalizeIndexOutput(
+      let outputSrcPathPage =
         srcRoute == null
           ? outputPathPageOrig
           : path.posix.join(
               entryDirectory,
               srcRoute === '/' ? '/index' : srcRoute
-            ),
-        isServerMode
-      );
+            );
+
+      if (!isAppPathRoute && !experimentalPPR) {
```

Comment:
if we're setting `isAppPathRoute` to true when `experimentalPPR` is true, could this check just be `!isAppPathRoute`?